providers/logingov_test.go

```diff
@@ -0,0 +1,134 @@
+package providers
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/stretchr/testify/assert"
+)
+
+func newLoginGovRedeemServer(body []byte) (*url.URL, *httptest.Server) {
+	s := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		rw.Write(body)
+	}))
+	u, _ := url.Parse(s.URL)
+	return u, s
+}
+
+func newLoginGovProvider() (l *LoginGovProvider, err error) {
+	keyData, err := ioutil.ReadFile("sample_key")
```

Comment:
Is it possible to generate a key on the fly rather than using a stored key within the repository? I worry that having example keys checked in may lead someone to use it somewhere where they shouldn't be, WDYT?

Reply:
It probably is.  I tried to do that at one point and found it to be hard in some way that I cannot remember, but you are right.  It would be better that way, so I'll give it another try.  :-)